shell/platform/embedder/embedder.h

```diff
@@ -989,6 +1087,27 @@ typedef struct {
   };
 } FlutterMetalBackingStore;
 
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterVulkanBackingStore).
+  size_t struct_size;
+  /// The VkImage handle that the layer will be rendered to. This image must
+  /// already be available for the engine to bind for writing when it's given to
+  /// the engine via the backing store creation callback. The engine will
+  /// perform a host sync for all layers prior to calling the compositor present
+  /// callback, and so the written layer images can be freely bound by the
+  /// embedder without any additional synchronization.
+  FlutterVulkanImageHandle image;
```

Comment:
This field pair is identical to `FlutterVulkanImage`. And I think it makes sense that we would need to extend that in the future (mip-levels, sample counts, usage, etc..). I think this should be replaced with a pointer to a const `FlutterVulkanImage` (pointer because adding members to that struct will not lead to ABI breaks in `FlutterVulkanBackingStore`).

Reply:
Oh, that's a great idea. Done.